src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,142 @@
+            ),
+            'default': '1121547',
+        },
+        {
+            'id': 'install_id',
+            'label': 'Installation id',
+            'type': 'string',
+            'help_text': gettext_noop(
+                'The  Installation ID from the GitHub App installation generated by the GitHub Admin '
+                'Example: 59980338 '
+                'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                'required for creating a limited github app token',
+            ),
+            'default': '59980338',
+        },
+        {
+            'id': 'ssh_key_data',
+            'label': gettext_noop('RSA Private Key'),
+            'type': 'string',
+            'format': 'ssh_private_key',
+            'secret': True,
+            'multiline': True,
+            'help_text': gettext_noop(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation ids.',
+            ),
+        },
+    ],
+    'required': [
+        'app_id',
+        'install_id',
+        'ssh_key_data',
+    ],
+}
+
+github_app_token = ManagedCredentialType(
+    namespace='github_app_token',
+    # scm source control management so it can be used for github
+    # authentication.
+    kind='scm',
+    name=gettext_noop('GitHub App Token'),
+    managed=True,
+    inputs=github_app_inputs,
+    custom_injectors=github_app_injector,
+)
+
+
+def github_app_backend(**kwargs):
+
+    github_url = kwargs.get('github_url')
+    app_id = kwargs.get('app_id')
+    installation_id = kwargs.get('install_id')
+    private_key = kwargs.get('ssh_key_data')
+    jwt_expiry = kwargs.get('jwt_expiry', 600)
+
+    missing_parameters = []
+    if not github_url:
+        missing_parameters.append('GitHub URL')
+    if not installation_id:
+        missing_parameters.append('Installation ID')
+    if not app_id:
+        missing_parameters.append('Application ID')
+    if not private_key:
+        missing_parameters.append('Private Key')
+    if missing_parameters:
+        raise Exception(f'Missing Parameter: {missing_parameters}')
+
+    # verify the installation id andn the app id are integers
+    try:
+        app_id = int(app_id)
+        installation_id = int(installation_id)
+    except ValueError:
+        raise Exception('App ID and Installation ID must be integers')
+
+    try:
+        jwt_expiry = int(jwt_expiry)
+    except ValueError:
+        raise Exception('JWT Expiry must be an integer')
+
+    auth = Auth.AppAuth(
+        app_id, private_key, jwt_expiry=jwt_expiry,
+    ).get_installation_auth(installation_id, None)
+
+    # In order to generate a token we have to make an initial call
+    # to GitHub even though we dont need to interact with GitHub.
+    Github(auth=auth)  # type: ignore
+
+    return auth.token
+
+
+github_app_lookup = CredentialPlugin(  # type: ignore
```

Comment:
Don't ignore unspecified rules, so this doesn't suppress any future violations automatically. Always be specific about the error code and include a justification comment.